app/src/main/java/eu/kanade/tachiyomi/source/online/english/Mangadex.kt

```diff
@@ -324,6 +326,101 @@ open class Mangadex(override val lang: String, private val internalLang: String,
         return manga
     }
 
+    fun fetchFollows(filters: FilterList) = fetchFollows(0, filters)
+
+    private fun fetchFollows(page: Int, filters: FilterList): Observable<List<Pair<SManga, Int>>> {
+        return getFollowsClient(filters).newCall(followsListRequest(page, filters))
+                .asObservable()
+                .map { response ->
+                    followsParse(response)
+                }.flatMap { parsedResponse ->
+                    // TODO I have no idea if this will work properly or if there's a better way
+                    val currentObservable = Observable.just(parsedResponse.first)
+                    if (parsedResponse.second)
+                        currentObservable.mergeWith(fetchFollows(page + 1, filters))
+                    else
+                        currentObservable
+                }
+    }
+
+    fun followsParse(response: Response): Pair<List<Pair<SManga, Int>>, Boolean> { // TODO exit generic hell
+        val document = response.asJsoup()
+
+        val follows = document.select(followSelector()).map { element ->
+            followFromElement(element)
+        }
+
+        val hasNextPage = followsNextPageSelector()?.let { selector ->
+            document.select(selector).first()
+        } != null
+
+        return Pair(follows, hasNextPage)
+    }
+
+    private fun getFollowsClient(filters: FilterList): OkHttpClient { // TODO reuse/rename `getSearchClient()`?
+        filters.forEach { filter ->
+            when (filter) {
+                is R18 -> {
+                    return when (filter.state) {
+                        1 -> clientBuilder(ALL)
+                        2 -> clientBuilder(ONLY_R18)
+                        3 -> clientBuilder(NO_R18)
+                        else -> clientBuilder()
+                    }
+                }
+            }
+        }
+        return clientBuilder()
+    }
+
+    fun followsListRequest(page: Int, filters: FilterList): Request {
+
+        // Format `/follows/manga/$status[/$sort[/$page]]`
+        val url = HttpUrl.parse("$baseUrl/follows/manga/0")!!.newBuilder() // Gets regardless of follow status. TODO add filter?
```

Comment:
url seems wrong.  If you passing in page.  Why 0 hardcoded

Reply:
or it like titles where 0 is English?

Reply:
Omitting `0` would return only follows with status "Reading"

Reply:
I had been wondering whether filters would be practical

Reply:
Actually it would be easier to filter in app from entire list vs filter through the request to mangadex